shard/committee/assignment.go

```diff
@@ -117,7 +117,12 @@ func eposStakedCommittee(
 	// TODO benchmark difference if went with data structure that sorts on insert
 	for i := range candidates {
 		validator, err := stakerReader.ReadValidatorInformation(candidates[i])
```

Comment:
Just realized this part may have potential problem. This ReadValidatorInformation is reading from the latest snapshot of the validator. Which means the eposStakedCommittee and the Compute() func will only work if it's called for the head of the chain. It won't work if it's called to compute for a history epoch.

Reply:
But I just checked the codebase, seems we never run Compute to calculate an epoch that's in the history, so it's good for now.